### PR TITLE
Fix/14100 lut table x overflow

### PR DESCRIPTION
**Pull request by zeeklop**
Fix overflow of table on X axis on Lookup tables, Caches and Data Adaptors Overview views

/nocl

## Description

Fix UI issue with tables in Lookup Tables views were overflowing the view port in the X axis.

## Motivation and Context

Fix UI bug

## How Has This Been Tested?

Manually

## Screenshots (if appropriate):
Data Adaptors Overview:
image

Caches Overview
image

Lookup Tables Overview
image

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by kingzacko1 — APPROVED** [review]
Fix for the issue at hand looks good. Can't speak for the JSX -> TSX conversions though.

**Review by simonychuang — APPROVED** [review]
Looks good, but since this is going into the core repo maybe it would be good to wait for a review from someone on that team.

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [inline graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:99]: Let's use `useCallback` for `onPageChange `, `onSearch`, `onReset` since this component has many children?

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [inline graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:112]: Looks like `buildHelpPopover` does not have to be a function call and it can just be the element.

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [inline graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:145]: I think it makes more sense to adjust the type definition for `withPaginationQueryParameter`. If we type it like this:
```
const withPaginationQueryParameter = <C extends React.ComponentType<React.ComponentProps<C>>>(Component: C, obj?: PaginationQueryParameterObject) => {
```

We do not need to use `as` when we use `withPaginationQueryParameter` for a function component.

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [reply graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:99]: Similar case in to `DataAdaptersOverview` and `LookupTablesOverview`.

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [reply graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:112]: Similar case in to `DataAdaptersOverview` and `LookupTablesOverview`.

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (linuspahl) [reply graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:145]: This would also be helpful in `DataAdaptersOverview` and `LookupTablesOverview`.

**Review by linuspahl — COMMENTED** [review]
Looks good to me and works as expected, just added some smaller suggestions.

**Comment by linuspahl** [comment]
Thank you for migrating the files to TS!

File: graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx
Comment (zeeklop) [reply graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:112]: So there is a weird issue with Overlay component, If we pass a component to the overlay prop, the overlay isn't properly positioned.

As a component:

image

As render function

image

Comment (linuspahl) [reply graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx:112]: I see, alright!

**Review by linuspahl — APPROVED** [review]

**Pull request merged**
